onnx/helper.py

```diff
@@ -333,7 +333,8 @@ def make_tensor(
         assert not raw, "Can not use raw_data to store string type"
 
     # Check number of vals specified equals tensor size
-    expected_size = 1 if (not raw) else (mapping.TENSOR_TYPE_TO_NP_TYPE[data_type].itemsize)
+    storage_type = mapping.TENSOR_TYPE_TO_STORAGE_TENSOR_TYPE[data_type]
+    expected_size = 1 if (not raw) else (mapping.TENSOR_TYPE_TO_NP_TYPE[storage_type].itemsize)
```

Comment:
The given raw type here should be the "storage type". For instance:
- float16: TensorProto.UINT16
- bfloat16: TensorProto.UINT16

This bug was uncaught by float16 because `np.dtype('float16').itemsize == np.dtype('uint16').itemsize == 2`.